ui/admin/config/environment.js

```diff
@@ -9,28 +9,36 @@ const APP_NAME = process.env.APP_NAME || 'Boundary';
 const API_HOST = process.env.API_HOST || '';
 const EDITION = process.env.EDITION || 'oss'; // Default edition is OSS
 
-// Object that defines edition features.
-const featureEditions = {
-  oss: {
-    'static-credentials': true,
-    'json-credentials': true,
-    byow: true,
-    'byow-pki-hcp-cluster-id': false,
-    'byow-pki-upstream': true,
-    'vault-worker-filter': false,
-    'target-worker-filters-v2': true,
-    'target-worker-filters-v2-ingress': false,
-    'target-worker-filters-v2-hcp': false,
-    'target-network-address': true,
-    'credential-library-vault-ssh-certificate': false,
-  },
+// Base edition declares available features, disabled by default.
+const baseEdition = {
+  byow: false,
+  'byow-pki-hcp-cluster-id': false,
+  'byow-pki-upstream': false,
+  'json-credentials': false,
+  'ssh-target': false,
+  'static-credentials': false,
+  'target-worker-filters-v2': false,
+  'target-worker-filters-v2-ingress': false,
+  'target-worker-filters-v2-hcp': false,
+  'target-network-address': false,
+  'vault-worker-filter': false,
+};
+// Editions maps edition keys to their associated featuresets.
+const featureEditions = {};
+featureEditions.oss = {
+  ...baseEdition,
+  byow: true,
+  'byow-pki-upstream': true,
+  'json-credentials': true,
+  'static-credentials': true,
+  'target-worker-filters-v2': true,
+  'target-network-address': true,
 };
 featureEditions.enterprise = {
   ...featureEditions.oss,
```

Comment:
Should we be inheriting from `oss`? I thought the idea of having a base was so we would only be turning features on, rather than turning any off if they're mutually exclusive. 

Currently we have one flag `byow-pki-upstream` that is mutually exclusive and would still need to set to false for  hcp

Reply:
`byow-pki-upstream` and `byow-pki-hcp-cluster-id` are currently the only mutually exclusive flags.  This could be refactored.

There isn't really a "base" edition.  A more apt name might be _available features_.  It just declares which flags exist.  Logically, enterprise does indeed "inherit" its features from OSS.  Ent and HCP are supersets (mostly).

Reply:
Address in #1658